tests/python/test_genotype_extractor_worker.py

```diff
@@ -0,0 +1,144 @@
+                             lambda v: v['info:SSC'],
+                             lambda v: float(v['info:GPV']),
+                             lambda v: float(v['info:SPV'])]
+        assert_keys_in_variants(expected_variants, variants,
+                                expected_extractors=expected_extractors,
+                                actual_extractors=actual_extractors)
+
+        # Make sure we've got the samples we expect (namely, TUMOR & NORMAL)
+        asserts.eq_(set(vcfreader.samples),
+                    set([v['sample_name'] for v in variants]))
+
+        # Make sure all the NORMAL & TUMOR samples have the same fields with the same values
+        def _ensure_samples_equivalent(sample_name):
+            def _get_sample_data(attr):
+                def _get_from_variant(variant):
+                    sample = find(variant.samples, lambda x: x.sample == sample_name)
+                    return getattr(sample.data, attr)
+                return _get_from_variant
+            expected_extractors = [_get_sample_data('GT'),
+                                   _get_sample_data('GQ'),
+                                   _get_sample_data('DP'),
+                                   _get_sample_data('RD'),
+                                   _get_sample_data('AD'),
+                                   _get_sample_data('FREQ'),
+                                   lambda v: tuple(_get_sample_data('DP4')(v))]
+            actual_extractors = [lambda v: v['sample:GT'],
+                                 # GQ is actually None on all variants in snv.vcf
+                                 lambda v: int(v['sample:GQ']) if v['sample:GQ'] else None,
+                                 lambda v: int(v['sample:DP']),
+                                 lambda v: int(v['sample:RD']),
+                                 lambda v: int(v['sample:AD']),
+                                 lambda v: v['sample:FREQ'],
+                                 lambda v: tuple(ast.literal_eval(v['sample:DP4']))]
+            tumor_variants = [v for v in variants
+                              if v['sample_name'] == sample_name]
+            assert_keys_in_variants(expected_variants, tumor_variants,
+                                    expected_extractors=expected_extractors,
+                                    actual_extractors=actual_extractors)
+
+        _ensure_samples_equivalent('NORMAL')
+        _ensure_samples_equivalent('TUMOR')
+
+
+
+def assert_keys_in_variants(
+        expected_variants, actual_variants,
+        expected_extractors=[], actual_extractors=[]):
+    """Asserts that the variants in in expected and actual match each other on
+    chromosome, position, reference, alternates, and the result of applying the
+    functions in expected_extractors and actual_extractors to expected_variants
+    and actual_variants, respectively.
+    """
+    def extract_expected_keys(v):
+        base = [v.CHROM, v.POS, v.REF, ''.join(map(str, v.ALT))]
```

Comment:
Wow, it took me a long time to understand why you were using `v.CHROM` here `v['contig']` down there. Maybe renaming `v` here to `pyvcf_variant` and `v` below to `db_variant`, or something to that effect?